### PR TITLE
add test for clone of non-empty message

### PR DESCRIPTION
**Pull request by elharo**
@jtattermusch as best I can tell UnknownFieldSet does not have a non-private method to read a field

### CODE REVIEW

**Comment by jtattermusch** [comment]
@elharo what is the context and why is this test important?

File: csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs [hunk 1]
```diff
@@ -138,6 +138,30 @@ public void TestClone(IMessage message)
             Assert.AreEqual(message.ToByteArray(), otherEmptyMessage.ToByteArray());
         }
 
+        [Test]
+        public void TestClone_lengthDelimited()
```
Comment (jtattermusch) [inline csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs:142]: nit: rename to TestClone_LengthDelimited()

File: csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs [hunk 2]
```diff
@@ -138,6 +138,30 @@ public void TestClone(IMessage message)
...
+        {
+            UnknownFieldSet destUnknownFieldSet = new UnknownFieldSet();
```
Comment (jtattermusch) [inline csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs:144]: nit: use `var destUnknownFieldSet = UnknownFieldSet()` instead of repeating the type name twice.  Also below.

File: csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs (hunk 1 above)
Comment (jtattermusch) [reply csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs:142]: Also, from the name it's not fully clear what this is actually supposed to test.

Comment (elharo) [reply csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs:142]: done

File: csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs (hunk 2 above)
Comment (elharo) [reply csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs:144]: done

File: csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs [hunk 3]
```diff
@@ -138,6 +138,29 @@ public void TestClone(IMessage message)
             Assert.AreEqual(message.ToByteArray(), otherEmptyMessage.ToByteArray());
         }
 
+        [Test]
+        public void TestClone_LengthDelimited()
+        {
+            var destUnknownFieldSet = new UnknownFieldSet();
```
Comment (jtattermusch) [inline csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs:144]: I feel that currently the test is hard to read. How about:

```
            var unknownVarintField = new UnknownField();
            unknownVarintField.AddVarint(99);
            
            var unknownLengthDelimitedField1 = new UnknownField();
            unknownLengthDelimitedField1.AddLengthDelimited(ByteString.CopyFromUtf8("some data"));

            var unknownLengthDelimitedField2 = new UnknownField();
            unknownLengthDelimitedField2.AddLengthDelimited(ByteString.CopyFromUtf8("some more data"));

            var destUnknownFieldSet = new UnknownFieldSet();
            destUnknownFieldSet.AddOrReplaceField(997, unknownVarintField);
            destUnknownFieldSet.AddOrReplaceField(999, unknownLengthDelimitedField1);
            destUnknownFieldSet.AddOrReplaceField(999, unknownLengthDelimitedField2);  // replace the field
```

Comment (elharo) [reply csharp/src/Google.Protobuf.Test/UnknownFieldSetTest.cs]: done

**Review by jtattermusch — COMMENTED** [review]
LGTM
